core/src/HIP/Kokkos_HIP_ZeroMemset.hpp

```diff
@@ -26,9 +26,10 @@ namespace Impl {
 template <class T, class... P>
 struct ZeroMemset<HIP, View<T, P...>> {
   ZeroMemset(const HIP& exec_space, const View<T, P...>& dst) {
-    KOKKOS_IMPL_HIP_SAFE_CALL(hipMemsetAsync(
-        dst.data(), 0, dst.size() * sizeof(typename View<T, P...>::value_type),
-        exec_space.hip_stream()));
+    KOKKOS_IMPL_HIP_SAFE_CALL(
+        (exec_space.impl_internal_space_instance()->hip_memset_async_wrapper(
+            dst.data(), 0,
+            dst.size() * sizeof(typename View<T, P...>::value_type))));
   }
 };
```

Comment:
Please do this in a separate PR, as multi-GPU support is not required for this change.

Also, please consider doing it as @dalg24 did in #7226.

Reply:
This pull request needs rebasing anyway.

Reply:
This was resolved in the rebase.